aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -95,10 +96,12 @@ module aptos_framework::transaction_validation {
                 || txn_sequence_number > 0
         ) {
             assert!(account::exists_at(transaction_sender), error::invalid_argument(PROLOGUE_EACCOUNT_DOES_NOT_EXIST));
-            assert!(
-                txn_authentication_key == account::get_authentication_key(transaction_sender),
-                error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
-            );
+            if (!features::simulation_enhancement_enabled() || !is_simulation || !is_auth_key_empty(&txn_authentication_key)) {
```

Comment:
Maybe it's just me but I find !(a && b && c) simpler to reason about than (!a || !b || !c) hahah.

Reply:
Thank you! I refactored it with a helper function.